builtin/credential/aws/path_login.go

```diff
@@ -996,16 +1001,19 @@ func (b *backend) pathLoginRenewIam(ctx context.Context, req *logical.Request, d
 	// renew existing tokens.
 	if roleEntry.InferredEntityType != "" {
 		if roleEntry.InferredEntityType == ec2EntityType {
-			instanceID, ok := req.Auth.Metadata["inferred_entity_id"]
-			if !ok {
-				return nil, fmt.Errorf("no inferred entity ID in auth metadata")
+			instanceID, err := getMetadataValue(req.Auth, "inferred_entity_id")
+			if err != nil {
+				return nil, err
 			}
-			instanceRegion, ok := req.Auth.Metadata["inferred_aws_region"]
-			if !ok {
-				return nil, fmt.Errorf("no inferred AWS region in auth metadata")
+			instanceRegion, err := getMetadataValue(req.Auth, "inferred_aws_region")
+			if err != nil {
+				return nil, err
 			}
-			_, err := b.validateInstance(ctx, req.Storage, instanceID, instanceRegion, req.Auth.Metadata["account_id"])
+			accountID, err := getMetadataValue(req.Auth, "account_id")
```

Comment:
Previously it doesn't seem like we were returning an error if account id wasn't set. Is this going to cause issues with older leases?

Reply:
Hm, good point. Theoretically _should_ have errored if it didn't exist, because [here](https://github.com/hashicorp/vault/blob/master/builtin/credential/aws/path_login.go#L1007) it would have been passed in as `""` to `validateInstance`. However, in tracing the code around, it does look _possible_ in some places that it could continue even with an account ID of `""`. Since there's not any test coverage either way, yes, I think that's a good idea - I will not error here and will pass in `""` if it's unset. Good catch.

Reply:
https://github.com/hashicorp/vault/pull/8991/commits/bb3cdb6d0231fa4e77c6a9ddbf7025d316d8530b

I put it at Debug log level versus Warn so if anyone was in this situation, it wouldn't spam their logs unless they wanted it to. Also, if there's a resulting error, it'll be loud enough without a preceding warning.